include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -823,6 +823,40 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+// Utility to recommend a stride for the best-performing memory access pattern from empirical testing on different
+// devices. This utility can only be called from the device.
+//
+// SPIR-V compilation targets show best performance with a stride of the sub-group size.
+// Other compilation targets perform best with a work-group size stride.
+template <typename NdItem>
+std::tuple<std::size_t, std::size_t, std::size_t, bool>
+__stride_recommender(const NdItem& __ndi, std::size_t __count, std::size_t __iters_per_work_item,
```

Comment:
I propose to rename `__ndi` to `__nd_item` or `__item`

Reply:
I have changed the name to `__item` here and in the submitter code as well.